File: VocalTrainerAndroid/Logic/src/main/java/com/neborosoft/vx/logic/SingingCompletionFlowBridge.kt
package com.neborosoft.vx.logic

import com.neborosoft.annotations.CppClass
import com.neborosoft.annotations.Releasable

@CppClass(withNativeConstructor = true)
interface SingingCompletionFlowBridge : Releasable {
    fun tryAgain()
    fun save()
    fun listen()
}

File: VocalTrainerAndroid/Logic/src/main/cpp/SingingCompletionFlowBridge.cpp
#include "SingingCompletionFlowBridge.h"

void SingingCompletionFlowBridge::listen() {
    cpp->listen();
}

void SingingCompletionFlowBridge::save() {
    cpp->save();
}

void SingingCompletionFlowBridge::tryAgain() {
    cpp->tryAgain();
}

SingingCompletionFlowBridge::~SingingCompletionFlowBridge() {
    delete cpp;
}

SingingCompletionFlowBridge::SingingCompletionFlowBridge(SingingCompletionFlow *cpp) : cpp(cpp) {}
